UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs

```diff
@@ -745,10 +908,89 @@ private void SaveData()
 			Logger.Log(JsonConvert.SerializeObject(ExternalCustomisationStorage), Category.Character);
 
 			PlayerManager.CurrentCharacterSettings = currentCharacter;
-			ServerData.UpdateCharacterProfile(
-				currentCharacter); // TODO Consider adding await. Otherwise this causes a compile warning.
+			// TODO Consider adding await. Otherwise this causes a compile warning.
+			ServerData.UpdateCharacterProfile(currentCharacter);
+			SaveCharacters();
+		}
+
+		/// <summary>
+		/// Takes 4 pictures of the character from all sides and stores them in %AppData%/Locallow/unitystation
+		/// </summary>
+		private IEnumerator<WaitForEndOfFrame> SaveCurrentCharacterSnaps()
+		{
+			savingPictures = true;
+			Util.CaptureUI capture = snapCapturer.GetComponent<Util.CaptureUI>();
+			int dir = 0;
+			capture.Path = $"/{currentCharacter.Username}/{currentCharacter.Name}"; //Note, we need to add IDs for currentCharacters later to avoid characters who have the same name overriding themselves.
+			while(dir < 4)
+			{
+				RightRotate();
+				capture.FileName = $"{currentDir}_{currentCharacter.Name}.PNG";
+				yield return WaitFor.EndOfFrame;
+				capture.TakeScreenShot();
+				dir++;
+			}
+			savingPictures = false;
+		}
+
+		/// <summary>
+		/// Remembers what was the last character the player chose in the character selector screen.
+		/// </summary>
+		private void SaveLastCharacterIndex()
+		{
+			PlayerPrefs.SetInt("lastCharacter", currentCharacterIndex);
+			PlayerPrefs.Save();
 		}
 
+		/// <summary>
+		/// Save all characters in a json file.
+		/// </summary>
+		private void SaveCharacters()
+		{
+			var settings = new JsonSerializerSettings
+			{
+				PreserveReferencesHandling = PreserveReferencesHandling.All,
+				NullValueHandling = NullValueHandling.Ignore,
+				ReferenceLoopHandling = ReferenceLoopHandling.Serialize,
+				Formatting = Formatting.Indented
+			};
+			SaveLastCharacterIndex(); //Remember the current character index, prevents a bug for newly created characters.
+			string json = JsonConvert.SerializeObject(PlayerCharacters, settings);
+			string path = Application.persistentDataPath + "characters.json";
+			if(File.Exists(path))
+			{
+				File.Delete(path);
+			}
+			File.WriteAllText(path, json);
+		}
+
+		/// <summary>
+		/// Get all characters that are saved in %APPDATA%/Locallow/unitystation/characters.json
```

Comment:
assume this changes per operating system

Reply:
https://docs.unity3d.com/ScriptReference/Application-persistentDataPath.html